arangod/Indexes/IndexFactory.cpp

```diff
@@ -752,6 +752,63 @@ Result IndexFactory::enhanceJsonIndexFulltext(VPackSlice definition,
   return res;
 }
 
+namespace {
+Result processIndexSortedPrefixFields(VPackSlice definition,
+                                      VPackBuilder& builder, size_t minFields,
+                                      size_t maxFields, bool create,
+                                      bool allowSubAttributes) {
+  TRI_ASSERT(builder.isOpenObject());
+
+  Result res;
+
+  auto fieldsSlice = definition.get(StaticStrings::IndexPrefixFields);
+
+  // storedValues are fully optional
+  if (!fieldsSlice.isNone()) {
+    if (fieldsSlice.isArray()) {
+      res = IndexFactory::validateFieldsDefinition(
+          definition, StaticStrings::IndexStoredValues, minFields, maxFields,
+          allowSubAttributes, /*allowIdAttribute*/ true);
+      if (res.ok() && fieldsSlice.length() > 0) {
+        std::unordered_set<std::string_view> fields;
+        for (VPackSlice it : VPackArrayIterator(fieldsSlice)) {
+          fields.insert(it.stringView());
+        }
+        auto normalFields = definition.get(StaticStrings::IndexStoredValues);
+        TRI_ASSERT(normalFields.isArray());
+        for (VPackSlice it : VPackArrayIterator(normalFields)) {
+          if (!fields.insert(it.stringView()).second) {
+            res.reset(TRI_ERROR_BAD_PARAMETER,
+                      "duplicate attribute name (overlap between index sorted "
+                      "prefix fields "
+                      "and index "
+                      "stored values list)");
```

Comment:
nit: we can as well break after the first error.
```suggestion
            res.reset(TRI_ERROR_BAD_PARAMETER,
                      "duplicate attribute name (overlap between index sorted "
                      "prefix fields "
                      "and index "
                      "stored values list)");
            break;
```